src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -19113,50 +19113,43 @@ void Unit::GetPartyMembers(std::list<Unit*>& TagUnitMap)
     }
 }
 
-Aura* Unit::AddAura(uint32 spellId, Unit* target)
+Aura* Unit::AddAura(uint32 aura)
 {
-    if (!target)
-        return nullptr;
-
-    SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(spellId);
+    SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(aura);
     if (!spellInfo)
         return nullptr;
 
-    if (!target->IsAlive() && !spellInfo->HasAttribute(SPELL_ATTR0_PASSIVE) && !spellInfo->HasAttribute(SPELL_ATTR2_ALLOW_DEAD_TARGET))
+    if (!IsAlive() &&
+        !spellInfo->HasAttribute(SPELL_ATTR0_PASSIVE) &&
+        !spellInfo->HasAttribute(SPELL_ATTR2_ALLOW_DEAD_TARGET))
         return nullptr;
 
-    return AddAura(spellInfo, MAX_EFFECT_MASK, target);
-}
-
-Aura* Unit::AddAura(SpellInfo const* spellInfo, uint8 effMask, Unit* target)
-{
-    if (!spellInfo)
-        return nullptr;
-
-    if (target->IsImmunedToSpell(spellInfo))
+    if (IsImmunedToSpell(spellInfo))
         return nullptr;
 
+    uint8 effectMask = MAX_EFFECT_MASK;
     for (uint32 i = 0; i < MAX_SPELL_EFFECTS; ++i)
     {
-        if (!(effMask & (1 << i)))
+        if (!(effectMask & (1 << i)))
             continue;
-        if (target->IsImmunedToSpellEffect(spellInfo, i))
-            effMask &= ~(1 << i);
-    }
 
-    if (Aura* aura = Aura::TryRefreshStackOrCreate(spellInfo, effMask, target, this))
-    {
-        aura->ApplyForTargets();
-        return aura;
+        if (IsImmunedToSpellEffect(spellInfo, i))
+            effectMask &= ~(1 << i);
     }
-    return nullptr;
+
+    Aura* auraObject = Aura::TryRefreshStackOrCreate(spellInfo, effectMask, this, this);
```

Comment:
This looks incorrect.

Ref https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Spells/Auras/SpellAuras.cpp#L326

Previously "caster" was the object casting the aura. Now its the same as the one receiving it.


The caster is relevant here: 
https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Entities/Unit/Unit.cpp#L4305
And here:
https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Spells/Auras/SpellAuras.cpp#L409



If the aura is not singlestack
https://github.com/azerothcore/azerothcore-wotlk/blob/06525cdc1e79cc041583af40b9f923adb7996d64/src/server/game/Entities/Unit/Unit.cpp#L4324
Then it does matter who the caster is, that information is lost in this refactor.

I haven't  checked what the caster info stored in Aura class is used for. But this alrady looks wrong.

Reply:
Thanks for the feedback.
I should have tagged my PR as a draft. I am well aware of the issues but the bigger problem is none of this code is elegant, performant or concise and the 'owner' in this context is completely irrelevant.